examples/lighting-app/linux/ui.cpp

```diff
@@ -104,6 +109,39 @@ class DeviceState
 
 DeviceState gDeviceState;
 
+/*
+ * Converts HSV with assumption that V == 100% into a IMGui color vector
+ */
+ImVec4 HueSaturationToColor(float hueDegrees, float saturationPercent)
+{
+    saturationPercent /= 100.0f;
+
+    float x = saturationPercent * static_cast<float>(1 - fabs(fmod(hueDegrees / 60, 2) - 1));
+
+    if (hueDegrees < 60)
+    {
+        return ImVec4(saturationPercent, x, 0, 1.0f);
+    }
+    if (hueDegrees < 120)
+    {
+        return ImVec4(x, saturationPercent, 0, 1.0f);
+    }
+    if (hueDegrees < 180)
```

Comment:
Considering use if else instead of multiple if 

if (x == 0)
{
    // Something
    x = 1;
}
else if (x == 1)
{
    // Something else...
}

This explicitly indicates only one of "Something" or "Something else..." will occur.

Reply:
Since I return every time, that would trigger https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html (in theory ... we probably are not running it on this file, however I generally agree with the premise for it).